projects/dot-rules/src/lib/semantic/modules/dropdown/dropdown.ts

```diff
@@ -1,163 +1,133 @@
+            [placeholder]="placeholder"
+            [options]="dropdownOptions | async"
+            [editable]="allowAdditions"
+            [(ngModel)]="modelValue"
+            *ngIf="maxSelections <= 1"
+            (onChange)="fireChange($event.value)"
+        >
+        </p-dropdown>
+        <dot-autocomplete-tags
+            [inputId]="name"
+            *ngIf="maxSelections > 1"
+            [value]="modelValue"
+            [options]="dropdownOptions | async"
+            [placeholder]="placeholder"
+            (onChange)="fireChange($event)"
+        >
+        </dot-autocomplete-tags>
+    `
 })
-export class Dropdown implements ControlValueAccessor {
-  @Input() name: string;
-  @Input() placeholder: string;
-  @Input() allowAdditions: boolean;
-  @Input() minSelections: number;
-  @Input() maxSelections: number;
-
-  @Output() onDropDownChange: EventEmitter<any> = new EventEmitter();
-  @Output() touch: EventEmitter<any> = new EventEmitter();
-  @Output() enter: EventEmitter<boolean> = new EventEmitter(false);
-
-  modelValue: string;
-
-  private _optionsAry: InputOption[] = [];
-  private _options: BehaviorSubject<InputOption[]>;
-  private elementRef: ElementRef;
-
-  onChange: Function = () => {};
-  onTouched: Function = () => {};
-
-  constructor(
-    elementRef: ElementRef,
-    @Optional() control: NgControl,
-    private loggerService: LoggerService
-  ) {
-    if (control && !control.valueAccessor) {
-      control.valueAccessor = this;
+export class Dropdown implements ControlValueAccessor, OnChanges {
+    @Input()
+    set value(value: string) {
+        this.modelValue = value;
     }
-    this.placeholder = '';
-    this.allowAdditions = false;
-    this.minSelections = 0;
-    this.maxSelections = 1;
-    this._options = new BehaviorSubject(this._optionsAry);
-    this.elementRef = elementRef;
-  }
-
-  @Input()
-  set value(value: string) {
-    this.modelValue = value;
-  }
-
-  get options() {
-    return this._options;
-  }
-
-  writeValue(value: any): void {
-    this.modelValue = _.isEmpty(value) ? '' : value;
-  }
-
-  registerOnChange(fn): void {
-    this.onChange = fn;
-  }
-
-  registerOnTouched(fn): void {
-    this.onTouched = fn;
-  }
-
-  fireChange(value: any): void {
-    this.modelValue = value;
-    if (this.onDropDownChange) {
-      this.onDropDownChange.emit(value);
-      this.onChange(value);
-      this.fireTouch(value);
+    @Input() options: any;
+    @Input() name: string;
+    @Input() placeholder: string;
+    @Input() allowAdditions: boolean;
+    @Input() minSelections: number;
+    @Input() maxSelections: number;
+
+    @Output() onDropDownChange: EventEmitter<any> = new EventEmitter();
+    @Output() touch: EventEmitter<any> = new EventEmitter();
+    @Output() enter: EventEmitter<boolean> = new EventEmitter(false);
+
+    modelValue: string;
+    dropdownOptions: Observable<MenuItem[]>;
+
+    constructor(@Optional() control: NgControl) {
```

Comment:
what's `@Optional` for?

Reply:
I didn't know this: https://angular.io/api/core/Optional#description interesting.